### ISSUE TITLE
[Bug]: Video autoplays despite settings to Block Audio and Video

### ISSUE THREAD

**Issue by mactruck79** [mozilla-mobile/fenix#21127]
### Steps to reproduce

1. Set Settings -> Site Permissions -> Autoplay to "Block audio and video" with no exceptions
2. Open tab and navigate to Reuters.com
2. Scroll down page to section titled "Top Video News" without clicking on anything

### Expected behaviour

The video inside the Top Video News section should NOT automatically start 

### Actual behaviour

The video inside the Top Video News section does start playing automatically without user interaction.  

### Device name

Google Pixel 3a XL

### Android version

Android 11

### Firefox release type

Firefox

### Firefox version

91.4.0

### Device logs

no crash logs captured

### Additional information

Originally reported here:  https://support.mozilla.org/en-US/questions/1348707

This video has audio and playback controls so it is not an animated GIF.  Per above link, I have tried changing settings for Enhanced Tracking Protection without resolving the unwanted autoplay behavior.

**Comment by mactruck79** [comment]
I was able to reproduce this behavior on two other Android devices that have Firefox 91.4.0 installed:

Motorola XT1575, Android version 7.0

Samsung Galaxy Tab A7, Android version 11

**Comment by VitalyVPinchuk** [comment]
Hello there!
I was unable to reproduce the issue neither with Xiaomi Mi9T Pro (Android 11) nor Emulator API 31 on the "main" thread (4c30483c).

**Comment by mactruck79** [comment]
Are you viewing www.reuters.com or mobile.reuters.com?  The mobile.reuters.com URL does not contain the Top Video News section that displays the auto-play video.

Since creating this post my phone was upgraded to Firefox version 92.1.0 so I tried again and was able to reproduce the unwanted autoplay video this morning.

**Comment by VitalyVPinchuk** [comment]
I did it like this

https://user-images.githubusercontent.com/26341194/132252604-b58ceeeb-f8e3-414f-b1f7-8219c3a3aa4e.mp4

**Comment by mactruck79** [comment]
That is odd, here is a video of what I see when I view the same URL with the same settings:

https://user-images.githubusercontent.com/72952663/132256556-d2b5e44c-9f3d-437d-a575-6daf7446a71f.mp4

**Comment by mactruck79** [comment]
Reproduced this behavior again on my Samsung Galaxy Tab A7, Android 11:

https://user-images.githubusercontent.com/72952663/132283398-d53895eb-3a09-4a75-a1d1-c7e5ef3b10b7.mp4

**Comment by Amejia481** [comment]
👋🏽  @mactruck79 thanks for reporting, would you mind trying on Firefox [nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix)? It will be helpful for us to see if the issue is reproduce-able there.

**Comment by mactruck79** [comment]
I tried this again after the recent mainstream upgrade to 92.1.1 and it appears to be resolved after toggling the Block Autoplay setting.  I took a video showing this behavior on my Tab A7, but I saw it happen first on my Pixel 3aXL last night.


https://user-images.githubusercontent.com/72952663/132915200-ca49d670-eda1-4517-b464-f7204e34ad60.mp4

I installed Firefox Nightly on Motorola XT1575 through the Google Play store and when I change Autoplay settings to Block Audio and Video and browse to reuters.com the Top Video News section does NOT autoplay.  I didn't get video of it, but the video autoplayed in Firefox 92.1.1, then I installed Nightly and it autoplayed there too - no audio.  I changed the setting to block Video and loaded a new tab with reuters.com.  After doing that I could not reproduce the behavior in both Nightly 94.0a1 and 92.1.1 applications.
